kong/router/atc_compat.lua

```diff
@@ -524,7 +549,9 @@ local function new_from_previous(routes, is_traditional_compatible, old_router)
       r.seen = nil
 
     else
-      assert(inst:remove_matcher(id))
+      if not inst:remove_matcher(id) then
```

Comment:
I think we can keep the `assert` for `remove_matcher`, it should really never fail and hiding it is not a good idea.

Reply:
done